### DISCUSSION TITLE
Resolver typed with type defintions generated by graphql-codegen is not callable.

### DISCUSSION THREAD

**Discussion by xoldd** [dotansimha/graphql-code-generator#8247] [Q&A]
I want to test a resolver function code. When I try to invoke that resolver typescript throws the following error :- `Cannot invoke an object which is possibly 'undefined'.`. I looked around and found an issue for this which has been fixed :- [fix](https://github.com/dotansimha/graphql-code-generator/issues/6310#issuecomment-932939901). I read the fix and added `makeResolverTypeCallable: true` to my `codegen.yml` file :- 
![Screenshot from 2022-08-17 20-32-05](https://user-images.githubusercontent.com/82873939/185174119-f400c96d-d0b0-42a7-a8bf-d4fb2fb7613a.png)
It's still throwing the same error. 
![Screenshot from 2022-08-17 20-34-34](https://user-images.githubusercontent.com/82873939/185174767-f04b1690-24b4-40bb-8f3a-d2517ec36604.png)
![Screenshot from 2022-08-17 20-35-02](https://user-images.githubusercontent.com/82873939/185174625-20cba5b2-aac8-4e3c-8a49-67874f6e6433.png)
![Screenshot from 2022-08-17 20-34-52](https://user-images.githubusercontent.com/82873939/185174630-179b2d4b-ab03-434f-aedf-83687e1e08a4.png)

 What's the fix?

**Comment by charlypoly** [comment]
Hi @xoldyckk,

Could you try using the [`avoidOptionals` configuration option](https://www.graphql-code-generator.com/plugins/typescript/typescript-resolvers)?
If you don't want to change your generated types, you call still update your call expression to the following: `post!()`
